### DISCUSSION TITLE
Is there any way to pass a struct which does not implement 'Copy' trait to a Component?

### DISCUSSION THREAD

**Discussion by AshWang-whm** [leptos-rs/leptos#2109] [Q&A]
I'm wondering if there is a way to pass a Component a struct that not implements the Copy trait?

![image](https://github.com/leptos-rs/leptos/assets/31700716/a2d8081b-12e3-4ed9-ac8e-4b84ed55ef28)

**Comment by gbj** [comment]
The `Fn` bound on `mount_to_body` is unnecessarily restrictive and should probably just be `FnOnce`. On the other hand, your code can simply be 
```rust
mount_to_body(|| {
  let obj = Parameter { /* etc */ };
  view! { <Component obj/> }
});
```
